core/src/main/java/com/linecorp/armeria/common/HttpData.java

```diff
@@ -104,7 +83,10 @@ static HttpData of(Charset charset, CharSequence text) {
 
         final CharBuffer cb = CharBuffer.wrap(text);
         final ByteBuffer buf = charset.encode(cb);
-        return of(buf.array(), buf.arrayOffset(), buf.remaining());
+
+        final byte[] bytes = new byte[buf.remaining()];
+        buf.get(bytes);
```

Comment:
We could avoid making a copy when `buf.arrayOffset()` is `0` and `buf.array().length` is `buf.remaining()`.